07.ls_object/run_ls.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+require_relative './file'
+require_relative './directory'
+
+opt = OptionParser.new
+
+params = { l: false, r: false, a: false }
+opt.on('-l') { |v| params[:l] = v }
+opt.on('-r') { |v| params[:r] = v }
+opt.on('-a') { |v| params[:a] = v }
+opt.parse!(ARGV)
+directory = ARGV[0] || '.'
+
+d = Directory.new(directory)
+
+d = d.reject_dot_files unless params[:a]
```

Comment:
```suggestion
d.reject_dot_files unless params[:a]
```

再代入しなくても`reject_dot_files`を呼べばインスタンスの状態（`@files`内のファイル）が変わるようになっていますね。
ただし、「状態がコロコロ変わっても自然なクラス」でなければ、なるべくインスタンスの状態は変えない方が良いです。

「状態がコロコロ変わっても自然なクラス」というのはたとえば「信号機クラス」などが考えられます。
信号機であれば赤、黄、緑という3つの状態があって、それが随時変わるというのは考え方として自然です。

一方、ディレクトリは実際にファイルを追加したり削除したりしないと状態（ファイルの件数）は変わらないですし、ディレクトリ自体が固定の並び順を持っているわけではない（ターミナル上で一度ソートしたらその後もずっとその状態が維持される、なんてことはない）ので、ディレクトリクラス内の`@files`が増えたり減ったり、並び順が変わったりするのは、「ディレクトリ」というオブジェクトを考えたときにあまり自然ではありません。

`reject_dot_files`や`reverse_files`はディレクトリというより、lsコマンドとして必要なメソッドだと考えた方が良いです。
（つまりディレクトリクラスの責務ではない）

Reply:
再代入の意味があまりない気がする…と思っていた部分でした。
ご指摘ありがとうございます。
ここは`display_files`として、ファイルの中身ではなく表示する内容を格納するようにしました。